Bug in textgrid renderer

### Describe the bug:
Bug in textgrid renderer when text contains some particular number of newlines. Tracked down to this repro.

### Example code:
```go
package main

import (
	"fyne.io/fyne"
	"fyne.io/fyne/app"
	"fyne.io/fyne/layout"
	"fyne.io/fyne/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("app")

	textGrid := widget.NewTextGrid()
	w.SetContent(
		fyne.NewContainerWithLayout(
			layout.NewGridWrapLayout(fyne.NewSize(200, 200)),
			textGrid,
		),
	)

	text := "\n\n\n\n\n\n\n\n\n\n\n\n"
	textGrid.SetText(text)
}

```


### Device:
 - Go version: 1.14.2
 - Fyne version: develop d7b5e3fad877

Sorry. App crashes:
```
panic: runtime error: index out of range [440] with length 440

goroutine 1 [running, locked to thread]:
fyne.io/fyne/widget.(*textGridRenderer).setCellRune(0xc00007c300, 0xc000000020, 0xdc, 0xa0a2a0, 0xc00000e3e0, 0x0, 0x0)
        /home/max/go/pkg/mod/fyne.io/fyne@v1.2.5-0.20200511094215-d7b5e3fad877/widget/textgrid.go:255 +0x3a7
fyne.io/fyne/widget.(*textGridRenderer).refreshGrid(0xc00007c300)
        /home/max/go/pkg/mod/fyne.io/fyne@v1.2.5-0.20200511094215-d7b5e3fad877/widget/textgrid.go:338 +0xff
fyne.io/fyne/widget.(*textGridRenderer).Refresh(0xc00007c300)
        /home/max/go/pkg/mod/fyne.io/fyne@v1.2.5-0.20200511094215-d7b5e3fad877/widget/textgrid.go:405 +0x2b
fyne.io/fyne/widget.(*BaseWidget).Refresh(0xc00003a600)
        /home/max/go/pkg/mod/fyne.io/fyne@v1.2.5-0.20200511094215-d7b5e3fad877/widget/widget.go:107 +0x51
fyne.io/fyne/widget.(*TextGrid).SetText(0xc00003a600, 0x963a54, 0xc)
        /home/max/go/pkg/mod/fyne.io/fyne@v1.2.5-0.20200511094215-d7b5e3fad877/widget/textgrid.go:106 +0x252
main.main()
        /home/max/Projects/go_test/textgrid.go:23 +0x178
exit status 2
```

--- a/widget/textgrid.go
+++ b/widget/textgrid.go
@@ -402,6 +402,7 @@ func (t *textGridRenderer) MinSize() fyne.Size {
 }
 
 func (t *textGridRenderer) Refresh() {
+	t.updateGridSize(t.text.size)
 	t.refreshGrid()
 }
 

--- a/widget/textgrid_test.go
+++ b/widget/textgrid_test.go
@@ -23,6 +23,16 @@ func TestNewTextGrid(t *testing.T) {
 
 func TestTextGrid_SetText(t *testing.T) {
 	grid := NewTextGrid()
+	grid.Resize(fyne.NewSize(20, 20))
+	text := "\n\n\n\n\n\n\n\n\n\n\n\n"
+	grid.SetText(text) // goes beyond the current view size - don't crash
+
+	assert.Equal(t, 13, len(grid.Rows))
+	assert.Equal(t, 0, len(grid.Rows[1].Cells))
+}
+
+func TestTextGrid_SetText_Overflow(t *testing.T) {
+	grid := NewTextGrid()
 	grid.SetText("Hello\nthere")
 
 	assert.Equal(t, 2, len(grid.Rows))

